scripts/e2e-test.sh

```diff
@@ -14,13 +14,159 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-set -euxo pipefail
+set -euo pipefail
+
+usage() {
+    cat <<EOT
+e2e-test.sh [-e|--env=(minikube|minishift)] [-p|--pause]
+
+Execute the end-to-end tests on a local minikube or minishift.
+
+-e|--env=(minikube|minishift) sets the environment the tests will be run under
+-p|--pause Pauses after each test step to help with debugging
+
+You can also specify the settings via environment variables (command line parameters take precedence).
+EUNOMIA_TEST_ENV=(minikube|minishift)
+EUNOMIA_TEST_PAUSE=yes
+
+EOT
+}
+
+function pause() {
+    if [[ "${EUNOMIA_TEST_PAUSE:-}" == "yes" ]]; then
+        read -r -s -n 1 -p "Press any key to continue . . ."
+        echo ""
+    fi
+}
+
+# Makes sure that all gitopsconfig jobs complete, before moving on to the next test
+# Requires the namespace name as the only parameter
+function wait_for_gitopsconfig_completion() {
+    NAMESPACE="${1}"
+    timeout=60
+    ALL_GOOD=0
+    JOBS=""
+    while ((--timeout)) && [[ "${ALL_GOOD}" == "0" ]]; do
```

Comment:
Can you add a comment above, the while (--timeout) will fail the test case if runs over timeout limit